### PR TITLE
40 create select date and time page

### LINKED ISSUE #40: Create Select Date and Time Page
The user will select the desired date and time to book a movie on this page. Dates are mapped from the current date to the next 11 dates (12 total). 

## Prerequisites

- [x] #19 

## Steps

- [x] Create the layout with the two sections (time, date)
- [x] Create the divider
- [x] format the dates with the date-fns library (they will most likely be coming in as ISOStrings)
- [x] Create a random function that makes three dates on random unavailable for the user to select
- [x] No time slots from past mapping from the current time should be bookable.
- [x] Use the already created Button Component

## Resources

- [Date-fns](https://date-fns.org/)

- https://www.figma.com/file/TOCJzJFMXE7PL1dQyGWqXZ/midterm?type=design&node-id=3%3A328&mode=design&t=poNoiZC2HH4cKdlU-1

### PR DESCRIPTION
**Pull request by pswu11**
closes #40 

- add select time page
- add ticket-store using zustand, and apply to select seat and ticket pages
- fix BookingDetails style, to prevent the text size from being over-written by `cn()`

Try out: start from /select-time, to /select-seat, to /ticket page, and see if the chosen seats and time are persistent.

### CODE REVIEW

**Comment by pswu11** [comment]
related to #41, will work with @adetuadetu

File: src/pages/SelectTime.tsx [hunk 1]
```diff
@@ -1,6 +1,188 @@
...
+      });
+    }
+  }
+  return timeslots;
+}
 
 export function SelectTime() {
-  return <p>Here we will create an awesome date & time picker</p>;
+  const [availableDates, setAvailableDates] = useState(
+    generateAvailableDates(8)
```
Comment (ReallyBoard123) [inline src/pages/SelectTime.tsx:85]: This has to be 12

File: src/pages/SelectTime.tsx [hunk 2]
```diff
@@ -1,6 +1,188 @@
...
+          >
+            {time.time}
+          </BookingDetails>
+        ))}
+      </div>
+      <Link
+        to={`/movies/${currentMovie.id}/select-seat`}
+        className="w-full mt-auto"
+      >
+        <Button className="w-full">Select Seat</Button>
```
Comment (ReallyBoard123) [inline src/pages/SelectTime.tsx:184]: Stop the user from going to the next page if they haven't selected the date and time. I apologize that I did not think it through before writing the issue.

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
That logic abstraction works out so nicely!! Great job.

File: src/pages/SelectTime.tsx (hunk 2 above)
Comment (pswu11) [reply src/pages/SelectTime.tsx]: good point!

**Pull request merged**